common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Agent/Default/Worker.cs

```diff
@@ -223,72 +220,66 @@ public class Worker : IWorker, IDisposable {
                         WorkerId, jobProcessInstruction.Job.Id);
                 }
                 catch (OperationCanceledException) {
-                    _logger.Information("Worker cancelled...");
+                    _logger.Information("Worker {workerId} cancelled...", WorkerId);
                 }
                 catch (Exception ex) {
                     // TODO: we should notify the exception
-                    _logger.Error(ex, "Worker: {Id}, exception during worker processing, wait {delay}...",
+                    _logger.Error(ex, "Worker {workerId}, exception during worker processing, wait {delay}...",
                         WorkerId, _jobCheckerInterval);
                     kModuleExceptions.WithLabels(AgentId, ex.Source, ex.GetType().FullName, ex.Message, ex.StackTrace, "Exception during worker processing").Inc();
                     await Task.Delay(_jobCheckerInterval, ct).ConfigureAwait(false);
                 }
             }
-            _logger.Information("Worker stopping...");
+            _logger.Information("Worker {workerId} stopping...", WorkerId);
         }
 
         /// <summary>
         /// Process jobs
         /// </summary>
-        /// <returns></returns>
         private async Task ProcessAsync(JobProcessingInstructionModel jobProcessInstruction,
             CancellationToken ct) {
             try {
                 // Stop worker heartbeat to start the job heartbeat process
                 _heartbeatTimer.Change(Timeout.InfiniteTimeSpan, Timeout.InfiniteTimeSpan); // Stop worker heartbeat
 
-                _logger.Information("Worker: {WorkerId} processing job: {JobId}, mode: {ProcessMode}",
+                _logger.Information("Worker {WorkerId} processing job {JobId}, mode: {ProcessMode}",
                     WorkerId, jobProcessInstruction.Job.Id, jobProcessInstruction.ProcessMode);
 
                 // Execute processor
                 while (true) {
-                    _jobProcess = null;
                     ct.ThrowIfCancellationRequested();
-                    using (_jobProcess = new JobProcess(this, jobProcessInstruction,
-                        _lifetimeScope, _logger)) {
-                        await _jobProcess.WaitAsync(ct).ConfigureAwait(false); // Does not throw
+                    if (_jobProcess == null) {
+                        _jobProcess = new JobProcess(this, jobProcessInstruction, _lifetimeScope, _logger);
                     }
+                    else {
+                        _jobProcess.ProcessNewInstruction(jobProcessInstruction);
+                    }
+                    await _jobProcess.WaitAsync(ct).ConfigureAwait(false); // Does not throw
```

Comment:
Is this operation going to complete at some point ? Or is it intended to wait until cancelled or reconfigured ?

Reply:
it's intended to be cancelled